### PR TITLE
IGVF-349-in-vitro-system

### PR DESCRIPTION
**Pull request by khzlin**

### CODE REVIEW

File: src/igvfd/schemas/in_vitro_system.json [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                "time_post_factors_introduction_units"
+            ],
+            "comment": "Biosample with a specified time_post_factors_introduction requires time_post_factors_introduction_units specification."
+        },
+        "time_post_factors_introduction_units": {
+            "required": [
+                "time_post_factors_introduction"
+            ],
+            "comment": "Biosample with a specified time_post_factors_introduction_units requires time_post_factors_introduction specification."
+        },
```
Comment (gabdank) [inline src/igvfd/schemas/in_vitro_system.json:67]: dependentRequired would be shorter and more straight forward

**Review by gabdank — CHANGES_REQUESTED** [review]

File: src/igvfd/schemas/in_vitro_system.json [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    "title": "In Vitro System",
+    "$id": "/profiles/in_vitro_system.json",
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "description": "Schema for submitting a sample grown in a in vitro system.",
+    "type": "object",
+    "required": [
+        "award",
+        "lab",
+        "source",
+        "donors",
```
Comment (gabdank) [inline src/igvfd/schemas/in_vitro_system.json:11]: use singular donor not plural

File: src/igvfd/schemas/in_vitro_system.json (hunk 1 above)
Comment (khzlin) [reply src/igvfd/schemas/in_vitro_system.json:67]: done

File: src/igvfd/schemas/in_vitro_system.json (hunk 2 above)
Comment (khzlin) [reply src/igvfd/schemas/in_vitro_system.json:11]: mixing still uses plural. Should I still changed it to singular?

Comment (khzlin) [reply src/igvfd/schemas/in_vitro_system.json]: changed to singular in other biosample and concrete samples as well.

File: src/igvfd/schemas/biosample.json [hunk 3]
```diff
@@ -205,7 +205,7 @@
                 "linkTo": "Treatment"
             }
         },
-        "donors": {
+        "donor": {
             "title": "Donors",
```
Comment (gabdank) [inline src/igvfd/schemas/biosample.json:209]: the title should be singular too

File: src/igvfd/schemas/in_vitro_system.json [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            "uniqueItems": true,
+            "items": {
+                "title": "Treatment",
+                "description": "A treatment applied to the biosample.",
+                "type": "string",
+                "linkTo": "Treatment"
+            }
+        },
+        "originated_from": {
+            "title": "Biosample Differentiated From",
```
Comment (gabdank) [inline src/igvfd/schemas/in_vitro_system.json:131]: the title is incorrect. it doesn't have to be differentiated from. we need something more genric. like originated from.

File: src/igvfd/schemas/in_vitro_system.json [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+            "description": "The biosample this organoid sample was differentiated from.",
```
Comment (gabdank) [inline src/igvfd/schemas/in_vitro_system.json:132]: description is incorrect, doesn't have to be differentiated

**Review by gabdank — CHANGES_REQUESTED** [review]

File: src/igvfd/schemas/in_vitro_system.json (hunk 4 above)
Comment (khzlin) [reply src/igvfd/schemas/in_vitro_system.json:131]: done

File: src/igvfd/schemas/in_vitro_system.json (hunk 5 above)
Comment (khzlin) [reply src/igvfd/schemas/in_vitro_system.json:132]: done

**Review by gabdank — DISMISSED** [review]

File: src/igvfd/schemas/in_vitro_system.json [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+                "week",
+                "month"
+            ]
+        },
+        "date_obtained": {
+            "title": "Date Collected"
+        },
+        "introduced_factors": {
+            "title": "Introduced Factors",
+            "description": "A list of treatments applied to the biosample that introduces factors.",
```
Comment (gabdank) [inline src/igvfd/schemas/in_vitro_system.json:119]: description: "A list of treatments applied to the biosample."

File: src/igvfd/schemas/in_vitro_system.json [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+        },
+        "time_post_factors_introduction": {
+            "title": "Time Post Factors Introduction",
+            "description": "The time that elapsed past the point when factors were introduced.",
+            "comment": "Use in conjunction with time_post_factors_introduction_units to specify time elapsed post factors introduction. Use of this property requires time_post_factors_introduction_units specification.",
+            "type": "number"
+        },
+        "time_post_factors_introduction_units": {
+            "title": "Time Post Factors Introduction Units",
+            "description": "The units of time that elapsed past the point when factors were introduced.",
```
Comment (gabdank) [inline src/igvfd/schemas/in_vitro_system.json:103]: "past the time-point when the factors"

File: src/igvfd/schemas/in_vitro_system.json [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+                "embryoid",
+                "cell line",
+                "differentiated cell",
+                "differentiated tissue",
+                "reprogrammed cell"
+            ]
+        },
+        "time_post_factors_introduction": {
+            "title": "Time Post Factors Introduction",
+            "description": "The time that elapsed past the point when factors were introduced.",
```
Comment (gabdank) [inline src/igvfd/schemas/in_vitro_system.json:97]: point -> "time-point"

File: src/igvfd/schemas/in_vitro_system.json [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+        ]
+    },
+    "properties": {
+        "schema_version": {
+            "default": "1"
+        },
+        "classification": {
+            "title": "Classification",
+            "description": "The general category of this type of in vitro system.",
+            "comment": "Required property. The specified category should match the ontology used to describe this type of in vitro system.",
```
Comment (gabdank) [inline src/igvfd/schemas/in_vitro_system.json:82]: "should match the ontology type used"

Comment (gabdank) [reply src/igvfd/schemas/in_vitro_system.json:82]: I think providing an example in the comment would be helpful as well. For example "brain organoid" using UBERON ontology.

Comment (gabdank) [reply src/igvfd/schemas/in_vitro_system.json:82]: cell line using CL ontology

**Review by gabdank — CHANGES_REQUESTED** [review]

File: src/igvfd/schemas/in_vitro_system.json (hunk 6 above)
Comment (khzlin) [reply src/igvfd/schemas/in_vitro_system.json:119]: done

File: src/igvfd/schemas/in_vitro_system.json (hunk 7 above)
Comment (khzlin) [reply src/igvfd/schemas/in_vitro_system.json:103]: done

File: src/igvfd/schemas/in_vitro_system.json (hunk 8 above)
Comment (khzlin) [reply src/igvfd/schemas/in_vitro_system.json:97]: done

File: src/igvfd/schemas/in_vitro_system.json (hunk 9 above)
Comment (khzlin) [reply src/igvfd/schemas/in_vitro_system.json:82]: done

**Review by gabdank — APPROVED** [review]

**Pull request merged**